main.gd

```diff
@@ -2,3 +2,11 @@ extends Node2D
 
 func _ready() -> void:
 	Input.set_mouse_mode(Input.MOUSE_MODE_HIDDEN)
+	var health = 100
+	var max_health = 100
+	var player = Player.create(health, max_health)
```

Comment:
should we have the Player `@export` a max_hp var that we can set to a default in the script but also editable in the scene editor if needed? and then init the UI health with the instantiated player's health values? i see they are _private but wasn't sure.

Reply:
The idea here is that the player itself doesn't know how much HP it should have, so this is providing it through outside.

- We know that we want things like Upgrades to determine how much hp or max_hp the "player" (or house) should have.
- At some point we might implement a save/load feature, and we probably don't want to clutter the player itself with that.

The inmediate need tho, is due to the HUD needing some initial value, and I didn't want the HUD to read directly from the player.

I have no clue about best practices tho, so I'm all ears if you have any other approaches.

Reply:
all good - let's see where this takes approach us